core/dns.go

```diff
@@ -191,20 +166,24 @@ func (dnsResolver *DNSResolver) LookupCAA(domain string, alias bool) ([]*dns.CAA
 	m := new(dns.Msg)
 	m.SetQuestion(dns.Fqdn(domain), dns.TypeCAA)
 
-	r, _, err := dnsResolver.LookupDNSSEC(m)
+	r, _, err := dnsResolver.ExchangeOne(m)
 	if err != nil {
 		return nil, err
 	}
 
 	var CAAs []*dns.CAA
-	for _, answer := range r.Answer {
-		if answer.Header().Rrtype == dns.TypeCAA {
-			caaR, ok := answer.(*dns.CAA)
-			if !ok {
-				err = errors.New("Badly formatted record")
-				return nil, err
+	// XXX: On resolver validation failure, or other server failures, return empty
+	//      set and no error.
+	if r.Rcode != dns.RcodeServerFailure {
+		for _, answer := range r.Answer {
+			if answer.Header().Rrtype == dns.TypeCAA {
+				caaR, ok := answer.(*dns.CAA)
+				if !ok {
+					err = errors.New("Badly formatted record")
+					return nil, err
+				}
+				CAAs = append(CAAs, caaR)
 			}
-			CAAs = append(CAAs, caaR)
 		}
```

Comment:
Would be more clear, re: above comment, if we returned `CAAs, nill` here since the function's nontrivial in length.